projects/orquestra-sdk/src/orquestra/sdk/_runtime/_ray/_client.py

```diff
@@ -91,10 +93,26 @@ def get(
             obj_refs: t.Union[ray.ObjectRef, t.List[ray.ObjectRef]],
             timeout: t.Optional[float] = None,
         ):
-            try:
-                return ray.get(obj_refs, timeout=timeout)
-            except (UserTaskFailedError, exceptions.GetTimeoutError, ValueError):
-                raise NotFoundError
+            loop_counter = 0
+            warning_printed = False
+            while True:
+                try:
+                    return ray.get(obj_refs, timeout=timeout)
+                except (UserTaskFailedError, exceptions.GetTimeoutError, ValueError):
+                    raise NotFoundError
+                except exceptions.RaySystemError as e:
+                    if loop_counter >= 20:
+                        raise
+                    else:
+                        # Ray seems to have a race condition in its code while getting
+                        # the task artifact.
+                        # https://github.com/ray-project/ray/issues/45027
+                        if not warning_printed:
+                            warning_printed = True
+                            warnings.warn(f"Ray get threw {e}. Retrying for 10s")
+                        time.sleep(0.5)
+                        loop_counter += 1
+                        continue
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏
What do you think of this? It doesn't print a warning, but we can extend the functionality of https://github.com/zapata-engineering/orquestra-sdk/blob/main/projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_retry.py to add that.

```python
        # Ray seems to have a race condition in its code while getting
        # the task artifact.
        # https://github.com/ray-project/ray/issues/45027
        @_retry.retry(attempts=20, allowed_exceptions=(exceptions.RaySystemError,), delay=0.5)
        def get(
            self,
            obj_refs: t.Union[ray.ObjectRef, t.List[ray.ObjectRef]],
            timeout: t.Optional[float] = None,
        ):
            try:
                return ray.get(obj_refs, timeout=timeout)
            except (UserTaskFailedError, exceptions.GetTimeoutError, ValueError):
                raise NotFoundError
```